gradio/themes/base.py

```diff
@@ -32,7 +46,179 @@ def repl_func(match):
         dark_css += "}"
         return css + "\n" + dark_css
 
+    def to_dict(self):
```

Comment:
I think we should add some unit tests for these new methods?

Reply:
Done!